File: app/src/main/java/com/example/administrator/essim/fragments/FragmentPixivRight.java
package com.example.administrator.essim.fragments;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.example.administrator.essim.R;
import com.example.administrator.essim.activities.ViewPagerActivity;
import com.example.administrator.essim.adapters.HotTagAdapter;
import com.example.administrator.essim.adapters.PixivAdapter;
import com.example.administrator.essim.models.PixivRankItem;
import com.example.administrator.essim.models.Tag;
import com.example.administrator.essim.utils.Common;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.sdsmdg.tastytoast.TastyToast;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.Response;


/**
 * Created by Administrator on 2018/1/15 0015.
 */

public class FragmentPixivRight extends BaseFragment {

    private Button mButton;
    private List<Tag> mTagList = new ArrayList<>();
    private HotTagAdapter mHotTagAdapter;
    private RecyclerView mRecyclerView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_pixiv_right, container, false);

        LinearLayoutManager mLinearLayoutManager = new LinearLayoutManager(mContext);
        mLinearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        mRecyclerView = view.findViewById(R.id.tag_list);
        mRecyclerView.setLayoutManager(mLinearLayoutManager);
        mRecyclerView.setHasFixedSize(true);
        getData("https://api.imjad.cn/pixiv/v1/?type=tags");
        return view;
    }


    private void getData(String address) {
        Common.sendOkhttpRequest(address, new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
                getActivity().runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        TastyToast.makeText(mContext, "数据加载失败", TastyToast.LENGTH_SHORT, TastyToast.CONFUSING).show();
                    }
                });
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {
                String responseData = response.body().string();
                Gson gson = new Gson();
                final List<Tag> booksInfo = gson.fromJson(responseData, new TypeToken<List<Tag>>() {}.getType());
                mTagList = booksInfo.subList(0, 60);
                mHotTagAdapter = new HotTagAdapter(mTagList, getContext());
                mHotTagAdapter.setOnItemClickLitener(new HotTagAdapter.OnItemClickLitener() {
                    @Override
                    public void onItemClick(View view, int position) {

                    }

                    @Override
                    public void onItemLongClick(View view, int position)
                    {
                        ClipboardManager cm = (ClipboardManager) mContext.getSystemService(Context.CLIPBOARD_SERVICE);
                        ClipData mClipData = ClipData.newPlainText("Label", booksInfo.get(position).getName());
                        cm.setPrimaryClip(mClipData);
                        Toast.makeText(mContext, booksInfo.get(position).getName()+ " 已复制到剪切板~",
                                Toast.LENGTH_SHORT).show();
                    }
                });
                getActivity().runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        mRecyclerView.setAdapter(mHotTagAdapter);
                    }
                });
            }
        });
    }
}

File: app/src/main/java/com/example/administrator/essim/fragments/FragmentPixivRight.kt
package com.example.administrator.essim.fragments

import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.administrator.essim.R
import com.example.administrator.essim.activities.SearchResultActivity
import com.example.administrator.essim.adapters.TrendingtagAdapter
import com.example.administrator.essim.interf.OnItemClickListener
import com.example.administrator.essim.network.AppApiPixivService
import com.example.administrator.essim.network.RestClient
import com.example.administrator.essim.response.TrendingtagResponse
import com.example.administrator.essim.utils.GridItemDecoration
import com.example.administrator.essim.utils.Common
import com.example.administrator.essim.utils.DensityUtil
import kotlinx.android.synthetic.main.fragment_pixiv_right.*
import retrofit2.Call
import retrofit2.Callback

class FragmentPixivRight : BaseFragment() {

    private var mPixivAdapter: TrendingtagAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_pixiv_right, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
    }

    private fun initView() {
        val gridLayoutManager = GridLayoutManager(mContext, 3)
        mRecyclerView.layoutManager = gridLayoutManager
        mRecyclerView.setHasFixedSize(true)
        mRecyclerView.addItemDecoration(GridItemDecoration(
                3, DensityUtil.dip2px(mContext, 6.0f), true));
        no_data.setOnClickListener {
            getHotTags()
        }
    }

    fun getHotTags() {
        mProgressbar.visibility = View.VISIBLE
        no_data.visibility = View.INVISIBLE
        val call = RestClient()
                .retrofit_AppAPI
                .create(AppApiPixivService::class.java)
                .getIllustTrendTags(Common.getLocalDataSet().getString("Authorization", "")!!)
        call.enqueue(object : Callback<TrendingtagResponse> {
            override fun onResponse(call: Call<TrendingtagResponse>, response: retrofit2.Response<TrendingtagResponse>) {
                try {
                    if(response.body()!!.trend_tags.size > 0) {
                        mPixivAdapter = TrendingtagAdapter(response.body()!!.trend_tags, mContext)
                        mPixivAdapter!!.setOnItemClickListener(object : OnItemClickListener {
                            override fun onItemClick(view: View, position: Int, viewType: Int) {
                                val intent = Intent(mContext, SearchResultActivity::class.java)
                                intent.putExtra("what is the keyword", response.body()!!.trend_tags[position].tag)
                                mContext.startActivity(intent)
                            }

                            override fun onItemLongClick(view: View, position: Int) {
                                //长按标签，可以跳转至标签封面的出处
                                Common.getSingleIllust(mProgressbar, mContext,
                                        response.body()!!.trend_tags[position].illust.id.toLong())
                            }
                        })
                        mRecyclerView.adapter = mPixivAdapter
                        no_data.visibility = View.INVISIBLE
                        mRecyclerView.visibility = View.VISIBLE
                        mProgressbar.visibility = View.INVISIBLE
                    }
                    else{
                        mRecyclerView.visibility = View.INVISIBLE
                        no_data.visibility = View.VISIBLE
                        mProgressbar.visibility = View.INVISIBLE
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                    mRecyclerView.visibility = View.INVISIBLE
                    no_data.visibility = View.VISIBLE
                    mProgressbar.visibility = View.INVISIBLE
                }
            }

            override fun onFailure(call: Call<TrendingtagResponse>, throwable: Throwable) {
                no_data.visibility = View.VISIBLE
                mRecyclerView.visibility = View.INVISIBLE
                mProgressbar.visibility = View.INVISIBLE
                Common.showToast(mContext, getString(R.string.no_proxy));
            }
        })
    }
}
